### DISCUSSION TITLE
Can I copy the fmt() function to use in my personal projects?

### DISCUSSION THREAD

**Discussion by ziontee113** [L3MON4D3/LuaSnip#523] [Q&A] [answered]
Hi @L3MON4D3 , I'm unfamiliar to open source rules, licenses & practices. My question is Can I copy and use the `fmt()` function (to interpolate strings) to use in my personal projects? If I can, besides crediting the owner, is there any else I have to do?

Thank you very much!

**Comment by L3MON4D3** [comment]
I think in general it depends on the license, but I have no problem with code being reused.
@jedrzejboczar, fine with you too?

**Reply by jedrzejboczar** [reply]
Sure, no problem for me.

**Reply by ziontee113** [reply]
@jedrzejboczar thank you very much :smile:

**Comment by ziontee113** [comment]
>I think in general it depends on the license

Does this mean that I have to use a specific license? Or is it depends on the owner's license?

```
Apache License 2.0

Permissions
  Commercial use
  Modification
  Distribution
  Patent use
  Private use

Limitations
  Trademark use
  Liability
  Warranty 
```

I see all the permissions, but I really don't know what "use" means :smile:

**Comment by L3MON4D3** [comment, marked as answer]
I have no idea about legal stuff, but if who youre asking has no problem with your reusing of some code, you're probably good.
So use whatever license you like :D
